src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -107,6 +216,11 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
 
             foreach (var referenceItem in result)
             {
+                if (referenceItem?.Location is null || referenceItem.Text is null)
```

Comment:
When would this happen? I assume you saw a few instances of it happening is why I ask 😄

Reply:
When the [conversion from `JToken` to `VSReference[]` happens](https://github.com/dotnet/aspnetcore-tooling/pull/2112/files#diff-4d367dc180f9d8b2a85502e662441dc9R104) in most cases, the conversion (deserialization) would go through regardless of the type of the serialized object (ie. not actually a `VSReference[]`).

If the serialized value's properties didn't match a `VSReference`, then it would create a `VSReference` with `null` properties. This check is just some basic verification that we deserialized a `VSReference` item and not something else.